### PR TITLE
fix permission issue with regular user jt admins

### PR DESCRIPTION
**Pull request by jlmitch5**
regular users assigned jt admin could not add user jt roles for users they couldn't edit

### CODE REVIEW

**Comment by AlanCoding** [comment]
I'll let you triage this, but if I sort the permissions page by first name, it changes the total count

jt_permissions_sort

This is because the original list checks for only users where `is_superuser` is false, but that filter is lost once the list is sorted.

Back to the issue at hand here, this user does not have edit permission to any of those users in the list, so the fact that they are not greyed out looks correct.

**Comment by AlanCoding** [comment]
This doesn't seem to resolve the issue.

Here, I am org admin of the "Default" organization.

I add a user who is system auditor, but not in my organization. This is not allowed, because this could allow me to change the password of the user, and gain system auditor privilege, which would be a permissions leak.

org_permissions_again

The UI just doesn't help here.

See JSON for users list:

```json
        {
            "id": 59,
            "type": "user",
            "url": "/api/v2/users/59/",
            "related": {
                "admin_of_organizations": "/api/v2/users/59/admin_of_organizations/",
                "authorized_tokens": "/api/v2/users/59/authorized_tokens/",
                "roles": "/api/v2/users/59/roles/",
                "organizations": "/api/v2/users/59/organizations/",
                "access_list": "/api/v2/users/59/access_list/",
                "teams": "/api/v2/users/59/teams/",
                "tokens": "/api/v2/users/59/tokens/",
                "personal_tokens": "/api/v2/users/59/personal_tokens/",
                "credentials": "/api/v2/users/59/credentials/",
                "activity_stream": "/api/v2/users/59/activity_stream/",
                "projects": "/api/v2/users/59/projects/"
            },
            "summary_fields": {
                "user_capabilities": {
                    "edit": false,
                    "delete": false
                }
            },
            "created": "2018-10-16T12:34:33.441351Z",
            "username": "system_auditor_XxTEGrc9R8",
            "first_name": "Joe (姀ⴹ㹝擀똭閠훉徹銉㬀)",
            "last_name": "System Auditor (矋䌸禦誷㪸삒쓊哣쥗猟)",
            "email": "pLoJHZya@test.biz",
            "is_superuser": false,
            "is_system_auditor": true,
            "ldap_dn": "",
            "external_account": null,
            "auth": []
        },
```

It clearly tells me that I don't have permission to edit this user. In that case, the UI shouldn't present me the option to make the user either organization admin or organization member, because that will result in a 403.

Or at minimum, the UI should give me some text about the error message the API returned in this situation.

**Comment by jlmitch5** [comment]
@AlanCoding thank you for that scenario!  I realized that the way I was going about this change before was incorrect.  Now, I am basing disabling off of the object you are trying to add permissions to.  For non-organization, the user shows up enabled:

screen shot 2018-11-05 at 4 44 37 pm

and for organization, they show up disabled:

screen shot 2018-11-05 at 4 44 59 pm

**Comment by jlmitch5** [comment]
also, filed low priority issue for the changing badge count https://github.com/ansible/awx/issues/2602

File: awx/ui/client/src/access/rbac-multiselect/rbac-multiselect-list.directive.js [hunk 1]
```diff
@@ -39,6 +40,8 @@ export default ['addPermissionsTeamsList', 'addPermissionsUsersList', 'TemplateL
             delete list.actions;
             delete list.fieldActions;
 
+            console.log(scope.objectType);
```
Comment (jakemcdermott) [inline awx/ui/client/src/access/rbac-multiselect/rbac-multiselect-list.directive.js]: should remove this line

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

**Comment by AlanCoding** [comment]
That should solve the main issue here. It's a big problem that the users are greyed out in the JT list. If this is what you want for the immediate fix, then it looks good to me.

There's the thornier issue that you _can_ give the auditor workflow_admin_role, or similar org-resource-roles, and that I don't think the UI has exposed that ability yet. It's not super important compared to fixing the JT permissions.

**Review by jakemcdermott — APPROVED** [review]

**Pull request merged**